File: frontend/src/app/modules/admindashboard/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private http: HttpClient) { }

  public getLimitedQuestion(id: any){
    return this.http.get(`http://localhost:8080/question/quiz/${id}`)
  }

  public getAllQuestion(id: any){
    return this.http.get(`http://localhost:8080/question/quiz/all/${id}`)
  }

  public getQuestionById(id: any){
    return this.http.get(`http://localhost:8080/question/get/${id}`)
  }

  public addQuestion(question: any) {
    return this.http.post('http://localhost:8080/question/add', question)
  }

  public updateQuestion(question : any){
    return this.http.put('http://localhost:8080/question/update', question)
  }

  public deleteQuestion(id: any){
    return this.http.delete(`http://localhost:8080/question/delete/${id}`)
  }

}

File: backend/src/main/java/com/exam/backend/services/QuestionService.java
package com.exam.backend.services;

import com.exam.backend.entity.Question;
import com.exam.backend.entity.Quiz;

import java.util.Set;

public interface QuestionService {

    public Question addQuestion(Question question);

    public Question updateQuestion(Question Question);

    public Set<Question> getAllQuestion();

    public Question getQuestion(Long id);

    public Set<Question> getQuestionByQuiz(Quiz quiz);

    public void deleteQuestion(Long id);

    public Question get(Long id);

}
